# Hypothesis testing (Buyer ratio)

## Null hypothesis(H0)=All proportions are equal

## Alternative hypothesis=All proportions are not equal.

## Significence level=5%(0.05)

In [10]:
## loading laibaries
import numpy as np
import pandas as pd
import scipy
import scipy.stats as stats
from scipy.stats import chi2_contingency

In [2]:
## Loading the data set
df=pd.read_csv("BuyerRatio.csv")

In [3]:
df

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [6]:
## Dropping Observed Values 
df.drop(['Observed Values'],axis=1,inplace=True)

In [7]:
df

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [8]:
ot=np.array([[50,142,131,70],[435,1523,1356,750]])
ot

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [12]:
# Chi2 contengency independence test
chi2_contingency(ot) # o/p is (Chi2 stats value, p_value, df, expected obsvations)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [14]:
# Chi2 contengency independence test
chi2_contingency(ot) # o/p is (Chi2 stats value, p_value, df, expected obsvations)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

## Conclusion

## From above result p-value i.e 0.660 is grater than 0.05 so we accept null hypothesis.

## Hence,All proportions are equal !